src/test/java/harness/utils/ExternalBQDatasets.java

```diff
@@ -106,6 +113,33 @@ public static void grantWriteAccess(
     grantAccess(datasetRef, memberEmail, memberType, "WRITER");
   }
 
+  /** Grants a given group dataViewer role to the specified table. */
+  public static void grantReadAccessToTable(
+      String projectId, String datasetId, String tableId, String groupEmail) throws IOException {
+    BigQueryCow bigQuery = getBQCow();
+    Policy policy =
+        bigQuery
+            .tables()
+            .getIamPolicy(projectId, datasetId, tableId, new GetIamPolicyRequest())
+            .execute();
+    List<Binding> updatedBindings =
+        Optional.ofNullable(policy.getBindings()).orElse(new ArrayList<>());
+    updatedBindings.add(
+        new Binding()
+            .setRole("roles/bigquery.dataViewer")
+            .setMembers(ImmutableList.of("group:" + groupEmail)));
+    bigQuery
+        .tables()
+        .setIamPolicy(
+            projectId,
+            datasetId,
+            tableId,
+            new SetIamPolicyRequest().setPolicy(policy.setBindings(updatedBindings)))
+        .execute();
+    System.out.println(
+        "Grant dataViewer access to table " + tableId + " for proxy group: " + groupEmail);
```

Comment:
nit: "group email" instead of "proxy group", since this is not necessarily a proxy group.